vocab.py

```diff
@@ -20,9 +28,102 @@ class ValidDataSourceName(Enum):
+
+        :return: Name of enum value.
+        '''
+        return str(self.name)
+
+
+class DataSourceStatus:
+    '''
+    Each data source process must return an instance of this class, which tells the name of the
+    data source and the time at which data from it was last updated.
+    '''
+
+    def __init__(
+            self,
+            data_source_name: ValidDataSourceName = NotImplemented,
+            data_updated_at: datetime = NotImplemented
+    ) -> None:
+        '''
+        Create a data source status object with name and update timestamp.
+
+        :param data_source_name: `ValidDataSourceName` - Name of data source
+        :param data_updated_at: `datetime` - Time when data from the source was updated
+        '''
+        self._data_source_name: ValidDataSourceName
+        self._data_updated_at: datetime
+        self.set_data_source_name(data_source_name).set_data_updated_at(data_updated_at)
+
+    @property
+    def data_source_name(self) -> ValidDataSourceName:
+        '''Name of the data source'''
+        return self._data_source_name
+
+    @data_source_name.setter
+    def data_source_name(self, data_source_name: ValidDataSourceName = NotImplemented):
+        self.set_data_source_name(data_source_name)
+
+    @property
+    def data_updated_at(self) -> datetime:
+        '''Time when data from the source was updated'''
+        return self._data_updated_at
+
+    @data_updated_at.setter
+    def data_updated_at(self, data_updated_at: datetime):
+        self.set_data_updated_at(data_updated_at)
+
+    def set_data_source_name(
+            self,
+            data_source_name: ValidDataSourceName = NotImplemented
+    ) -> DataSourceStatus:
+        '''
+        Set name of data source.  Main logic and fluent interface.
+
+        :param data_source_name: `ValidDataSourceName` - Name of data source
+        :return: `DataSourceStatus`
+        '''
+        if (data_source_name is NotImplemented):
+            raise ValueError('data_source_name must be specified')
+        if (not isinstance(data_source_name, ValidDataSourceName)):
+            raise ValueError('data_source_name must be of type ValidDataSourceName')
+
+        self._data_source_name = data_source_name
+        return self
+
+    def set_data_updated_at(self, data_updated_at: datetime = NotImplemented) -> DataSourceStatus:
+        '''
+        Set time data source updated.  Main logic and fluent interface.
+
+        :param data_updated_at: `datetime` - Time data source updated
+        :return: `DataSourceStatus`
+        '''
+        if (data_updated_at is NotImplemented):
+            self._data_updated_at = pd.to_datetime(time.time(), unit='s', utc=True)
+        elif (data_updated_at.tzinfo is not pytz.UTC):
+            raise ValueError('_data_updated_at must have UTC time zone')
+        else:
+            self._data_updated_at = data_updated_at
+        return self
+
+    def copy(self) -> Dict:
```

Comment:
Nit: could be more precise here with the type hint, e.g. `Dict[str, Union[ValidDataSourceName, pd.Timestamp, datetime]]`

Reply:
Removed Pandas dependency.